doc/languages-frameworks/rust.md

```diff
@@ -261,6 +264,43 @@ general. A number of other parameters can be overridden:
   };
   ```
 
+### Setting Up `nix-shell`
+Oftentimes you want to develop code using `nix-env`. Using the example above,
+we want to do the following:
+
+- Have access to `cargo` and `rustc`
+- Have the `openssl` library available to the rust crate through
+  it's _normal_ compilation mechanism (`pkg-config`).
+
+`default.nix`:
+```
+with import <nixpkgs> {};
+((import ./hello.nix).hello {}).override {
+  crateOverrides = defaultCrateOverrides // {
+    hello = attrs: {
+        buildInputs = [
+        # Some example external dependencies
+        openssl xorg.libX11 xorg.libXtst
+        # For `nix-shell`
+        rustc cargo openssl pkgconfig
+      ];
+
+      # Set Relevant ENV Variables
+      PKG_CONFIG_PATH = "${openssl}/lib/pkgconfig";
```

Comment:
And neither should this.

Reply:
I'm frankly shocked. I arrived at the need of environment variables through a highly technical process of _trial and error_ (/sarcasm). I _swear_ this didn't work before... but now it totally does.

So... I would like to have _some_ environment variable example. I'm going to do `RUST_BACKTRACE` since that is a nice to have.